src/rpdk/core/contract/resource_client.py

```diff
@@ -361,22 +362,32 @@ def assert_time(self, start_time, end_time, action):
 
     @staticmethod
     def assert_primary_identifier(primary_identifier_paths, resource_model):
-        assert all(
-            traverse(resource_model, fragment_list(primary_identifier, "properties"))[0]
-            for primary_identifier in primary_identifier_paths
-        )
+        try:
+            assert all(
+                traverse(
+                    resource_model, fragment_list(primary_identifier, "properties")
+                )[0]
+                for primary_identifier in primary_identifier_paths
+            ), "Every model MUST include the primaryIdentifier"
+        except KeyError:
+            raise AssertionError("Every model MUST include the primaryIdentifier")
 
     @staticmethod
     def is_primary_identifier_equal(
         primary_identifier_path, created_model, updated_model
     ):
-        return all(
-            traverse(created_model, fragment_list(primary_identifier, "properties"))[0]
-            == traverse(updated_model, fragment_list(primary_identifier, "properties"))[
-                0
-            ]
-            for primary_identifier in primary_identifier_path
-        )
+        try:
+            return all(
+                traverse(
+                    created_model, fragment_list(primary_identifier, "properties")
+                )[0]
+                == traverse(
+                    updated_model, fragment_list(primary_identifier, "properties")
+                )[0]
+                for primary_identifier in primary_identifier_path
+            )
+        except KeyError:
+            raise AssertionError("Every model MUST include the primaryIdentifier")
```

Comment:
can we throw KeyError instead of assertion error?

Reply:
same as above, I dont think we should expose runtime errors and rather handle them accordingly (ideally have custom exception classes for each case like that)